main.go

```diff
@@ -125,6 +125,10 @@ func mainError() error {
 	daemonCommand.PersistentFlags().String(f.Service.Azure.HostCluster.ResourceGroup, "", "Host cluster resource group name.")
 	daemonCommand.PersistentFlags().String(f.Service.Azure.HostCluster.VirtualNetwork, "", "Host cluster virtual network name.")
 	daemonCommand.PersistentFlags().Bool(f.Service.Azure.MSI.Enabled, true, "Whether to enabled Managed Service Identity (MSI).")
+	daemonCommand.PersistentFlags().Int(f.Service.Azure.Network.CalicoSubnetMask, 27, "calico network subnet mask.")
+	daemonCommand.PersistentFlags().Int(f.Service.Azure.Network.MasterSubnetMask, 24, "master network subnet mask.")
+	daemonCommand.PersistentFlags().Int(f.Service.Azure.Network.VPNSubnetMask, 24, "vpn network subnet mask.")
+	daemonCommand.PersistentFlags().Int(f.Service.Azure.Network.WorkerSubnetMask, 24, "worker network subnet mask.")
```

Comment:
In general I don't like the design with the flags.

We have now following information in the CR

```
    virtualNetwork:
      calicoSubnetCIDR: 10.1.128.0/17
      cidr: 10.1.0.0/16
      masterSubnetCIDR: 10.1.0.0/24
      workerSubnetCIDR: 10.1.1.0/24
```

After this is merged we should pass only:

```
    virtualNetwork:
      cidr: 10.1.0.0/16
```

We shouldn't rely on fixed mask numbers here.

We always should take second half for calico (as kubernetesd does now).
Then we should assert that mask for `virtualNetwork.cidr` is > 22. And assign `/24` subnets for master, worker and VPN.

Does that make sense? If not we can have a call.

Reply:
so instead of calico /17, we should use `ipam.Half`

for the rest my intuition would tell me, let's ipam do the hard stuff and fails if it has too, there not much we can do there if the provided network is too small.

Reply:
Yes is mask is <22 this is unrecoverable.